src/components/MarkdownActionsDropdown/index.tsx

```diff
@@ -0,0 +1,120 @@
+import { Menu } from '@headlessui/react'
+import { IconCopy, IconEye, IconExternal } from '@posthog/icons'
+import { Chevron } from 'components/Icons'
+import CheckIcon from '../../images/check.svg'
+import React, { useState } from 'react'
+
+interface CopyMarkdownActionsDropdownProps {
+    /** The markdown content to work with */
+    markdownContent: string
+    /** Page URL for context */
+    pageUrl: string
+}
+
+export const CopyMarkdownActionsDropdown: React.FC<CopyMarkdownActionsDropdownProps> = ({
+    markdownContent,
+    pageUrl,
+}) => {
+    const [copiedState, setCopiedState] = useState<string | null>(null)
+
+    // Consolidated styles
+    const menuItemButtonStyles =
+        '!m-0 py-1.5 px-3 !text-sm cursor-pointer rounded-sm hover:bg-light active:bg-accent dark:hover:bg-light/10 dark:active:bg-light/5 transition-colors hover:transition-none whitespace-nowrap w-full text-left flex items-center'
+    const menuItemIconStyles = 'w-4 h-4 mr-3'
+
+    // Helper function to strip query parameters from URL
```

Comment:
No need to do this, the URL object is already able to do most of these utils

https://developer.mozilla.org/en-US/docs/Web/API/URL